org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/ast/StringTemplate.java

```diff
@@ -0,0 +1,122 @@
+/*******************************************************************************
+ * Copyright (c) 2023 IBM Corporation and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.internal.compiler.ast;
+
+import org.eclipse.jdt.internal.compiler.codegen.CodeStream;
+import org.eclipse.jdt.internal.compiler.codegen.ConstantPool;
+import org.eclipse.jdt.internal.compiler.flow.FlowContext;
+import org.eclipse.jdt.internal.compiler.flow.FlowInfo;
+import org.eclipse.jdt.internal.compiler.impl.Constant;
+import org.eclipse.jdt.internal.compiler.lookup.BlockScope;
+import org.eclipse.jdt.internal.compiler.lookup.TypeBinding;
+import org.eclipse.jdt.internal.compiler.lookup.TypeIds;
+import org.eclipse.jdt.internal.compiler.util.Util;
+
+public class StringTemplate extends Expression {
+	private StringLiteral[] fragments;
+	private Expression[] values;
+	public StringTemplate(StringLiteral[] fragments, Expression[] values, int start, int end) {
+		this.fragments = fragments;
+		this.values = values;
+		this.sourceStart = start;
+		this.sourceEnd = end;
+	}
+
+    public StringLiteral[] fragments() {
+		return this.fragments;
+    }
+    public Expression[] values() {
+		return this.values;
+    }
+	@Override
+	public void resolve(BlockScope scope) {
+		if (this.constant != Constant.NotAConstant) {
+			this.constant = Constant.NotAConstant;
+		}
```

Comment:
This looks suspicious -

Reply:
Perhaps, not required for the StringTemplate. But we need this for the TemplateExpression. The check was overly cautious; I will remove it.